Mage.Sets/src/mage/cards/a/AndurilFlameOfTheWest.java

```diff
@@ -0,0 +1,84 @@
+package mage.cards.a;
+
+import mage.abilities.Ability;
+import mage.abilities.common.AttacksAttachedTriggeredAbility;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.continuous.BoostEquippedEffect;
+import mage.abilities.keyword.EquipAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.Outcome;
+import mage.constants.SubType;
+import mage.constants.SuperType;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+import mage.game.permanent.token.SpiritWhiteToken;
+
+import java.util.UUID;
+
+/**
+ * @author LePwnerer
+ */
+public final class AndurilFlameOfTheWest extends CardImpl {
+
+    public AndurilFlameOfTheWest(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ARTIFACT}, "{3}");
+
+        this.supertype.add(SuperType.LEGENDARY);
+        this.subtype.add(SubType.EQUIPMENT);
+
+        // Equipped creature gets +3/+1
+        this.addAbility(new SimpleStaticAbility(new BoostEquippedEffect(3, 1)));
+
+        // Whenever equipped creature attacks, create two tapped 1/1 white Spirit creature tokens with flying.
+        // If that creature is legendary, instead create two of those tokens that are tapped and attacking.
+        this.addAbility(new AttacksAttachedTriggeredAbility(new AndurilFlameOfTheWestEffect()));
```

Comment:
you need to use the `setTargetPointer` argument here to make sure that the trigger keeps track of the right creature